Topics/Tech_Stacks/DigitalOcean_Web_App.md

```diff
@@ -0,0 +1,129 @@
+In digital ocean, a "Droplet" refers to a virtual private server (VPS) that you can deploy and manage on DigitalOcean's cloud infrastructure. It's essentially a scalable compute platform with add-on storage, where you can run applications, websites, and other services.
+
+1. Log in to your DigitalOcean account.
+2. On the left panel, click on "Droplets" under "Manage" and press "Create Droplet".
+
+Note that droplets do have a price per month to host on their platform, but DigitalOcean provides a free $200 budget for 60 days to newly created accounts.
+
+2. Choose a data center region. If you'd like the fastest speeds between your device and the web app, choose the closest region to you.
+3. Choose your operating system. Ubuntu is a popular choice for operating systems, as it is known to be fast.
+4. Select a plan based on your application's requirements. The CPUs and sizes that the platform offers differ in price.
+5. Optionally, you can also add an authentication method to your droplet to make it a private web app.
+6. Click "Create Droplet."
+
+<img src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/103388045/f45f232d-6eb4-4857-9e05-98e600d66791" height="500"/>
+
+### Step 2: Access Your Droplet
+Once the droplet is created, you will receive an email with login details.
+
+Use an SSH client to connect to your droplet. For example, run `ssh root@your_droplet_ip` in the terminal.
+
+### Step 3: Update System Packages
+After logging in, update the system packages:
+
+```bash
+   sudo apt update
+   sudo apt upgrade
+```
+
+This ensures that your droplet's operating system is up-to-date with the latest security patches and software updates.
+
+### Step 4: Install Dependencies
+Install any dependencies required for your web application, such as Node.js, Python, or database drivers.
+
+For example, if you're using Node.js, you could run the following command in the terminal:
+
+```bash
+   sudo apt install nodejs npm
+```
+
+Adjust the command based on the requirements of your specific application.
+
+### Step 5: Transfer your files
+Transfer your application files to the droplet using SCP or any other method. To do this with SCP, follow these steps:
+
+1. Open a new terminal and use the cd command to navigate to the directory containing your application code.
+
+```bash
+   cd /path/
+```
+
+2. Use the scp command to copy the files into the newly created droplet.
+
+```bash
+   scp -r your_application_directory user@droplet_ip:/remote/path
+```
+
+Replace the following:
+  your_application_directory: The local path to your application code.
+  user: The username used to log in to your droplet.
+  droplet_ip: The IP address of your droplet. You can get this on the webpage of the droplet.
+  /remote/path: The path on your droplet where you want to copy the files.
+
+3. Enter your password. You'll be prompted to enter the password using the login details from the email previously.
```

Comment:
Instead of previously, you could specify the email you received after signup to make it more clear, in case they receive multiple emails for promotional purposes or forgetting their password.